File: frontend/src/api/recetteArrosage.ts
import client from './client'

export interface RecetteArrosageLigne {
  id_ligne?:    number
  id_produit:   number
  quantite:     number
  unite?:       string   // g | Kg | mL | L
  note_ligne?:  string
  ordre:        number
  nom_produit?: string
  type_produit?: string
}

export interface RecetteArrosage {
  id_recette_arrosage: number
  nom_recette:   string
  type_arrosage?: string   // Eau simple | Eau + amendements
  quantite_eau?:  number
  unite_eau?:     string   // L | mL
  notes?:         string
  lignes:         RecetteArrosageLigne[]
}

export type RecetteArrosageCreate = Omit<RecetteArrosage, 'id_recette_arrosage'>
export type RecetteArrosageUpdate = Partial<RecetteArrosageCreate>

export const recetteArrosageAPI = {
  getAll:    ()                                              => client.get<RecetteArrosage[]>('/recettes-arrosage/'),
  getById:   (id: number)                                    => client.get<RecetteArrosage>(`/recettes-arrosage/${id}`),
  create:    (data: RecetteArrosageCreate)                   => client.post<RecetteArrosage>('/recettes-arrosage/', data),
  update:    (id: number, data: RecetteArrosageUpdate)       => client.put<RecetteArrosage>(`/recettes-arrosage/${id}`, data),
  delete:    (id: number)                                    => client.delete(`/recettes-arrosage/${id}`),
  exportCSV: ()                                              => client.get('/recettes-arrosage/export/csv', { responseType: 'blob' }),
  importCSV: (file: File) => {
    const form = new FormData(); form.append('file', file)
    return client.post('/recettes-arrosage/import', form, { headers: { 'Content-Type': 'multipart/form-data' } })
  },
}

File: backend/app/schemas/recette_arrosage.py
"""Schemas Pydantic pour RecetteArrosage"""
from pydantic import BaseModel, ConfigDict
from typing import Optional, List


class RecetteArrosageLigneBase(BaseModel):
    id_produit: int
    quantite:   float
    unite:      Optional[str] = None   # mL | L | g | Kg
    note_ligne: Optional[str] = None
    ordre:      int = 0


class RecetteArrosageLigneCreate(RecetteArrosageLigneBase):
    pass


class RecetteArrosageLigneRead(RecetteArrosageLigneBase):
    id_ligne:     int
    nom_produit:  Optional[str] = None
    type_produit: Optional[str] = None
    model_config  = ConfigDict(from_attributes=True)


class RecetteArrosageBase(BaseModel):
    nom_recette:   str
    type_arrosage: Optional[str]  = None   # Eau simple | Eau + amendements
    quantite_eau:  Optional[float]= None
    unite_eau:     Optional[str]  = None   # L | mL
    notes:         Optional[str]  = None


class RecetteArrosageCreate(RecetteArrosageBase):
    lignes: List[RecetteArrosageLigneCreate] = []


class RecetteArrosageUpdate(BaseModel):
    nom_recette:   Optional[str]  = None
    type_arrosage: Optional[str]  = None
    quantite_eau:  Optional[float]= None
    unite_eau:     Optional[str]  = None
    notes:         Optional[str]  = None
    lignes:        Optional[List[RecetteArrosageLigneCreate]] = None


class RecetteArrosageRead(RecetteArrosageBase):
    id_recette_arrosage: int
    lignes:              List[RecetteArrosageLigneRead] = []
    model_config         = ConfigDict(from_attributes=True)
